common/src/main/java/apoc/util/Util.java

```diff
@@ -335,17 +335,19 @@ public static URLConnection openUrlConnection(String url, Map<String, Object> he
         URL src = new URL(url);
         URLConnection con = src.openConnection();
         con.setRequestProperty("User-Agent", "APOC Procedures for Neo4j");
-        if (headers != null) {
-            Object method = headers.get("method");
-            if (method != null && con instanceof HttpURLConnection) {
+        if (con instanceof HttpURLConnection) {
                 HttpURLConnection http = (HttpURLConnection) con;
-                http.setRequestMethod(method.toString());
-                http.setChunkedStreamingMode(1024*1024);
-                http.setInstanceFollowRedirects(true);
+                http.setInstanceFollowRedirects(false);
```

Comment:
When is the redirection resolved?

Reply:
openUrlConnection is called recursively by readHttpInputStream.

```
public static StreamConnection readHttpInputStream(String urlAddress, Map<String, Object> headers, String payload) throws IOException {
        ApocConfig.apocConfig().checkReadAllowed(urlAddress);
        URLConnection con = openUrlConnection(urlAddress, headers); // CALLED HERE
        writePayload(con, payload);
        String newUrl = handleRedirect(con, urlAddress);
        if (newUrl != null && !urlAddress.equals(newUrl)) { // IF REDIRECT WILL MAKE IT IN HERE
            con.getInputStream().close();
            return readHttpInputStream(newUrl, headers, payload);
        }

        return new StreamConnection.UrlStreamConnection(con);
    }
```